src/main/java/com/powsybl/openloadflow/RemoteVoltageTargetChecker.java

```diff
@@ -0,0 +1,270 @@
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow;
+
+import com.google.common.base.Stopwatch;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.math.matrix.DenseMatrix;
+import com.powsybl.math.matrix.MatrixFactory;
+import com.powsybl.math.matrix.SparseMatrixFactory;
+import com.powsybl.openloadflow.ac.AcJacobianMatrix;
+import com.powsybl.openloadflow.ac.AcLoadFlowParameters;
+import com.powsybl.openloadflow.ac.equations.AcEquationSystemCreator;
+import com.powsybl.openloadflow.ac.equations.AcEquationType;
+import com.powsybl.openloadflow.ac.equations.AcVariableType;
+import com.powsybl.openloadflow.ac.solver.AcSolverUtil;
+import com.powsybl.openloadflow.adm.AdmittanceEquationSystem;
+import com.powsybl.openloadflow.adm.AdmittanceMatrix;
+import com.powsybl.openloadflow.equations.EquationSystem;
+import com.powsybl.openloadflow.equations.EquationTerm;
+import com.powsybl.openloadflow.equations.JacobianMatrix;
+import com.powsybl.openloadflow.equations.VariableSet;
+import com.powsybl.openloadflow.graph.EvenShiloachGraphDecrementalConnectivityFactory;
+import com.powsybl.openloadflow.graph.GraphConnectivityFactory;
+import com.powsybl.openloadflow.network.*;
+import com.powsybl.openloadflow.network.impl.LfNetworkLoaderImpl;
+import com.powsybl.openloadflow.network.util.UniformValueVoltageInitializer;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.jgrapht.Graph;
+import org.jgrapht.graph.Pseudograph;
+import org.jgrapht.traverse.BreadthFirstIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
+ */
+public class RemoteVoltageTargetChecker {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteVoltageTargetChecker.class);
+
+    private final LfNetwork network;
+
+    private final EquationSystem<AcVariableType, AcEquationType> equationSystem;
+
+    private final JacobianMatrix<AcVariableType, AcEquationType> j;
+
+    public RemoteVoltageTargetChecker(LfNetwork network,
+                                      EquationSystem<AcVariableType, AcEquationType> equationSystem,
+                                      JacobianMatrix<AcVariableType, AcEquationType> j) {
+        this.network = Objects.requireNonNull(network);
+        this.equationSystem = Objects.requireNonNull(equationSystem);
+        this.j = Objects.requireNonNull(j);
+    }
+
+    public static Set<String> findElementsToDiscardFromVoltageControl(Network network, LoadFlowParameters parameters) {
+        return findElementsToDiscardFromVoltageControl(network, parameters, new SparseMatrixFactory());
+    }
+
+    public static Set<String> findElementsToDiscardFromVoltageControl(Network network, LoadFlowParameters parameters, MatrixFactory matrixFactory) {
```

Comment:
Is it possible to add a reason ?
   Incompatible with XXX
   Irrealistic

(It seems at this time there is even no log to understand the discard recommendation)